pyxlsb/styles.py

```diff
@@ -25,14 +63,48 @@ def _parse(self):
         self._cell_styles = list()
         self._cell_style_xfs = list()
 
+        self._xf_record = dict()
+        self._format_record = dict()
+
         self._fp.seek(0, os.SEEK_SET)
         for rectype, rec in RecordReader(self._fp):
             # TODO
-            if rectype == rt.END_STYLE_SHEET:
+            if rectype == 47:
+                self._xf_record[len(self._xf_record) - 1] = rec
+            elif rectype == 44:
+                self._format_record[rec.fmtId] = rec
+            elif rectype == rt.END_STYLE_SHEET:
                 break
 
     def get_style(self, idx):
         return None
 
+    def get_number_format(self, idx):
+        if idx is None:
+            return FormatRecord(fmtId=0, fmtCode='General')
+        elif idx in self._xf_record:
+            numFmtId = self._xf_record[idx].numFmtId
+            if numFmtId in self._format_record:
+                return self._format_record[numFmtId]
+            elif numFmtId in self._default_format:
+                return FormatRecord(fmtId=numFmtId, fmtCode=self._default_format[numFmtId]["format"])
+        return FormatRecord(fmtId=0, fmtCode='General')
+
+    def get_dtype(self, idx):
```

Comment:
See my other note about moving this logic directly to the cell converted value handling.